### PR TITLE
Added end point to save access token without redirect via shimmer

### PR DESCRIPTION
**Pull request by saurabmitra**
Save AccessToken without shimmer redirect and use these tokens to pull metric data.

### CODE REVIEW

File: shim-server/src/main/java/org/openmhealth/shimmer/common/controller/LegacyAuthorizationController.java [hunk 1]
```diff
@@ -211,22 +237,35 @@ public AuthorizationRequestParameters initiateAuthorization(
...
+                throw new ShimException(
+                    "Error occurred redirecting to :" + authParams.getRedirectUri());
             }
             return null;
         }
 
         return ok(response);
     }
 
+    @PostMapping(value = "/authorize")
```
Comment (saurabmitra) [inline shim-server/src/main/java/org/openmhealth/shimmer/common/controller/LegacyAuthorizationController.java]: This is the end point and DTO that i have added. The rest are formatting edits.

Comment (karshit) [reply shim-server/src/main/java/org/openmhealth/shimmer/common/controller/LegacyAuthorizationController.java]: If this is completely independent from private methods, would suggest adding a new Controller class and inject `accessParametersRepo` there for your usage.

File: shim-server/src/main/java/org/openmhealth/shimmer/common/domain/AccessTokenDto.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package org.openmhealth.shimmer.common.domain;
+
+
+import javax.validation.constraints.NotNull;
+
+public class AccessTokenDto {
+
+    @NotNull
+    private long userId;
```
Comment (karshit) [inline shim-server/src/main/java/org/openmhealth/shimmer/common/domain/AccessTokenDto.java]: This validation is of no use due to usage of primitive `long`.
Move to `Long` instead.

File: shim-server/src/main/java/org/openmhealth/shimmer/common/domain/AccessTokenDto.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+    @NotNull
+    private String shim;
```
Comment (karshit) [inline shim-server/src/main/java/org/openmhealth/shimmer/common/domain/AccessTokenDto.java:13]: Use `@NotEmpty` or `@NotBlank` instead

File: shim-server/src/main/java/org/openmhealth/shimmer/common/controller/LegacyAuthorizationController.java (hunk 1 above)
Comment (saurabmitra) [reply shim-server/src/main/java/org/openmhealth/shimmer/common/controller/LegacyAuthorizationController.java]: All the '/authorize' URLs are handled by this controller. That's why have added this end point here. Should I change?

**Pull request merged**